tests/Helios.FsCheck.Tests/Channels/Sockets/Models/TcpServerSocketModelComparer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+
+namespace Helios.FsCheck.Tests.Channels.Sockets.Models
+{
+    /// <summary>
+    /// <see cref="IEqualityComparer{T}"/> that verifies that any two <see cref="ITcpServerSocketModel"/> are equivalent, regardless of the
+    /// order of messages received / written and the order of clients joined / not joined.
+    /// </summary>
+    public class TcpServerSocketModelComparer : IEqualityComparer<ITcpServerSocketModel>
```

Comment:
Compares the models, bearing in mind that the collections will be ordered differently.